xarray/test/__init__.py

```diff
@@ -215,6 +215,24 @@ def assertDataArrayAllClose(self, ar1, ar2, rtol=1e-05, atol=1e-08):
         self.assertVariableAllClose(ar1, ar2, rtol=rtol, atol=atol)
         self.assertCoordinatesEqual(ar1, ar2)
 
+def assert_equal(a, b):
+    import xarray as xr
+    ___tracebackhide__ = True
+    assert type(a) == type(b)
+    if isinstance(a, xr.DataArray):
+        assert_equal(a.data_vars, b.data_vars)
+        assert_equal(a.coords, b.coords)
+    elif isinstance(a, xr.Variable):
+        assert as_variable(a).equals(b), (a, b)
+    elif isinstance(a, xr.Dataset):
+        assert a.equals(b), (a, b)
```

Comment:
I think this (`assert a.equals(b), (a, b)`) would actually probably suffice for all of `Dataset`, `DataArray` and `Variable`.